WalletWasabi.Tests/UnitTests/Crypto/ZeroKnowledge/LinearRelationTests.cs

```diff
@@ -0,0 +1,109 @@
+using NBitcoin.Secp256k1;
+using System.Linq;
+using WalletWasabi.Crypto;
+using WalletWasabi.Crypto.Groups;
+using WalletWasabi.Crypto.ZeroKnowledge.LinearRelation;
+using Xunit;
+
+namespace WalletWasabi.Tests.UnitTests.Crypto.ZeroKnowledge
+{
+	public class EquationTest
+	{
+		[Theory]
+		[InlineData(0, 0)]
+		[InlineData(0, 1)]
+		[InlineData(1, 0)]
+		[InlineData(1, 1)]
+		[InlineData(1, 2)]
+		[InlineData(3, 5)]
+		[InlineData(5, 7)]
+		[InlineData(7, 11)]
+		[InlineData(short.MaxValue, uint.MaxValue)]
+		[InlineData(int.MaxValue, uint.MaxValue)]
+		[InlineData(uint.MaxValue, uint.MaxValue)]
+		public void VerifyResponsesAndSimulations(uint scalarSeed1, uint scalarSeed2)
+		{
+			var witness = new ScalarVector(new[] { new Scalar(scalarSeed1), new Scalar(scalarSeed2) });
```

Comment:
If you would like to change:

`var witness = new ScalarVector(new[] { new Scalar(scalarSeed1), new Scalar(scalarSeed2) });`

to

`var witness = new ScalarVector(new Scalar(scalarSeed1), new Scalar(scalarSeed2));`

then you need the following constructor:

```csharp
public ScalarVector(params Scalar[] scalars)
{
     Scalars = scalars;
}
```

Reply:
this is already done in #4280

Reply:
Yes, that looks good to me. Let's see how we use this class in the real world and then if this ctor is useful it will be self evident.